packages/app-project/src/screens/SubjectTalkPage/components/SubjectTalkData/components/Tags/TagsContainer.jsx

```diff
@@ -66,204 +66,271 @@ function TagsContainer({
-            const newData = prevData.map(t => {
+            return prevData.map(t => {
               if (t.id === tag.id) {
                 return {
                   ...t,
-                  userVoted: false,
                   vote_count: 0
                 }
               }
               return t
             })
-            return newData
           }
         },
-        { revalidate: false }
+        { revalidate: false } // don't need to revalidate any request to /tags/popular
       )
     } else {
-      // To remove a vote from a votable tag,
-      // update the votable tag to show no user vote and decrement the vote count
+      // Modify vote_count of the tag in the votableTags array
       mutateVotableTags(
         prevData => {
           if (!prevData) {
             return []
           } else {
-            const newData = prevData.map(t => {
+            return prevData.map(t => {
               if (t.id === tag.id) {
                 return {
                   ...t,
-                  userVoted: false,
                   vote_count: t.vote_count > 0 ? t.vote_count - 1 : 0
                 }
               }
               return t
             })
-            return newData
           }
         },
-        { revalidate: false }
+        { revalidate: false } // don't need to revalidate votableTags because optimistic data is enough
       )
     }
 
-    // Remove the tag vote from the user's tag votes
     mutateTagVotes(
-      prevData => {
+      async (prevData) => {
+        setVoteUpdating(true)
+        // remove the "tag vote" from the database and the database changes the relevant votableTag's vote_count to 0
+        await removeTagVote(tagVoteToRemove.id)
+        setVoteUpdating(false)
+
+        // this async function must return something to put into the tagVotes data array
+        // so remove the tag vote from the tagVotes array
         if (!prevData) {
           return []
         } else {
-          const newData = prevData.filter(vote => vote.id !== tagVoteToRemove.id)
-          return newData
+          return prevData.filter(
+            vote => vote.id !== tagVoteToRemove.id
+          )
         }
       },
-      { revalidate: false }
+      {
+        // optimistically remove the tag vote from the tagVotes array before the network request happens
+        optimisticData: prevData => {
+          if (!prevData) {
+            return []
+          } else {
+            return prevData.filter(
+              vote => vote.id !== tagVoteToRemove.id
+            )
+          }
+        },
+        // don't need to revalidate tagVotes because optimisticData is enough
+        revalidate: false,
+        // rollbackOnError will respond to an error thrown by removeTagVote()
+        // tagVotes will be rolled back, but the above mutations to popularTags or votableTags will not,
+        // this could be improved to rollback all mutations in handleRemoveVote()
+        rollbackOnError(err) {
```

Comment:
Because TagsContainer.jsx contains multiple `mutate` functions per handler after clicking a Tag, the `rollbackOnError` function only works for the mutation it belongs to. There could be some better handling of rollback UI, but this is enough for this PR. You can test this caught error by removing `authorization` from `removeTagVote()`.